pom.xml

```diff
@@ -170,7 +170,6 @@
                 <environment>
                   <os>win32</os>
                   <ws>win32</ws>
-                  <arch>x86_64</arch>
```

Comment:
Does this actually work? I thought that `<arch>` was a mandatory tag? If so you'd have to declare a whole new `<environment>` entry.

Reply:
Probably not, but I'll test that when the native build is working.

Reply:
@Phillipus what you suggest is quite logical, but here's the mystery: if I add a new `<environment>` entry for the `aarch64` Windows, the build for the repo will fail (when building in either x64 and/or Arm64 boxes), and I can't work out why. Perhaps @HannesWell can help debugging it.

So eventually I came up with the current solution: remove that `<arch>x86_64</arch>` line so that the POM file is applicable for both environments.

Reply:
I would guess that win32/win32/aarch64 triplet has to be defined in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/da0494c0d177b4e5f8d0d41a713c5359b2a2fa04/eclipse-platform-parent/pom.xml#L244 first.

Reply:
> I would guess that win32/win32/aarch64 triplet has to be defined in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/da0494c0d177b4e5f8d0d41a713c5359b2a2fa04/eclipse-platform-parent/pom.xml#L244 first.

Good point, created https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1917 for that.

I also updated the p2.inf to require the native fragment through P2 at a WoA and the local target-config activated on Windows in a profile. With the latter I'm not sure if the change mentioned above would be necessary just for SWT. In general it probably is.

Reply:
Yes, this `p2.inf` file is the cause of my build problems of this repo for over a year! Thanks @HannesWell for the fix! Who would have thought that an (seemingly) unimportant file in an obscure corner of the repo would have such an impact on the build? With this fix for `p2.inf`, my builds for this repo in both x64 and aarch64 Windows boxes now succeed! (with that new `win32/win32/aarch64` triplet added to the pom.xml)

> I would guess that win32/win32/aarch64 triplet has to be defined in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/da0494c0d177b4e5f8d0d41a713c5359b2a2fa04/eclipse-platform-parent/pom.xml#L244 first.

Nope, this is for stand-alone build of this repo, nothing to do with the aggregator yet. In fact, my plan would be to change the aggregator repo as the very last step, after changing all submodules first for WoA (like, this SWT submodule, then equinox, eclipse.platform, etc.)

Reply:
> Yes, this `p2.inf` file is the cause of my build problems of this repo for over a year! Thanks @HannesWell for the fix! Who would have thought that an (seemingly) unimportant file in an obscure corner of the repo would have such an impact on the build?

Yes sometimes it is difficult and the tiny details make the difference. I'm currently working on making that workaround obsolete but it is not yet fully ready. But I'm glad it is working now. :)